lorahw/gateway.c

```diff
@@ -154,46 +152,82 @@ int setUpGateway(int bus, int region) {
     lut.size = 16;
 
     if(lgw_txgain_setconf(0, &lut) != LGW_HAL_SUCCESS) {
-        return 6;
+        return 7;
     }
 
     // start the gateway.
-    int res = lgw_start();
+    int res = start_gateway();
     if (res != LGW_HAL_SUCCESS) {
-        return 7;
+        return 8;
     }
     return 0;
- }
+}
 
-int stopGateway() {
+int stop_gateway() {
     return lgw_stop();
 }
 
-struct lgw_pkt_rx_s* createRxPacketArray() {
+struct lgw_pkt_rx_s* create_rx_packet_array() {
     return (struct lgw_pkt_rx_s*)malloc(sizeof(struct lgw_pkt_rx_s) * MAX_RX_PKT);
 }
 
-int receive(struct lgw_pkt_rx_s* packet)  {
-    return lgw_receive(MAX_RX_PKT, packet);
+void disable_buffering() {
+    setbuf(stdout, NULL);
+}
+
+#ifdef TESTING
+void redirect_to_pipe(int fd) {
+    // Mock implementation for testing - does nothing
 }
 
 int send(struct lgw_pkt_tx_s* packet) {
-    return lgw_send(packet);
+    // for testing, return 0 - sucesssful
+    return 0;
 }
 
-void disableBuffering() {
-    setbuf(stdout, NULL);
+int start_gateway() {
+    // for testing, return no error
+    return 0;
 }
 
-#ifdef TESTING
-void redirectToPipe(int fd) {
-    // Mock implementation for testing - does nothing
+int receive(struct lgw_pkt_rx_s* packet) {
+    // testing - fill the packet with test values
+    packet->size = 3;
+    uint8_t payload[3] = {0x01, 0x02, 0x03};
+    memcpy(packet->payload, payload, packet->size);
+    packet->datarate = 7;
+    packet->snr = 20;
```

Comment:
didn't see this before, but does it mean anything that we set these values here?

Reply:
jk this is all testing which makes way more sense